src/moonlink/src/storage/mooncake_table/transaction_stream.rs

```diff
@@ -424,27 +423,28 @@ impl MooncakeTable {
         }
     }
 
-    pub async fn flush_stream(&mut self, xact_id: u32, lsn: Option<u64>) -> Result<()> {
-        let mut disk_slice = self.prepare_stream_disk_slice(xact_id, lsn)?;
-        self.flush_stream_disk_slice(xact_id, &mut disk_slice)
-            .await?;
-        if let Some(lsn) = lsn {
-            self.next_snapshot_task.new_flush_lsn = Some(lsn);
-        }
-        self.apply_stream_flush_result(xact_id, disk_slice);
+    pub fn flush_stream(&mut self, xact_id: u32, lsn: Option<u64>) -> Result<()> {
+        // Temporarily remove to drop reference to self
+        let mut stream_state = self
+            .transaction_stream_states
+            .remove(&xact_id)
+            .expect("Stream state not found for xact_id: {xact_id}");
```

Comment:
suggest to attach LSN as well, so ourselves could differentiate stream flush and stream commit